core/io/resource.cpp

```diff
@@ -600,7 +600,7 @@ void ResourceCache::clear() {
 }
 
 bool ResourceCache::has(const String &p_path) {
-	lock.lock();
+	MutexLock mutex_lock(lock);
```

Comment:
I'd suggest putting this one in a scope in a way that it's unlocked before the `!res! check, to better follow the (implicit) rule of unlocking as soon as possible. In this case, a simple null check won't keep the mutex locked for a lot longer, though, but for consistency and micro-pessimization avoidance, I'd go for that.